templates/curl_json.conf.erb

```diff
@@ -40,10 +40,22 @@ LoadPlugin "curl_json"
 <% if @header -%>
     Header "<%= @header %>"
 <% end -%>
+<%-
+  def valid_json?(json)
+    JSON.parse(json)
+    return true
+  rescue JSON::ParserError
+    return false
+  end
+-%>
 <% unless @post.nil? -%>
+<% if valid_json?(@post) -%>
+    Post "<%= @post.gsub('\\"','\\\\\\\\\"').gsub!(%r{(?<!\\)"},'\"') %>"
+<% else -%>
     Post "<%= @post %>"
```

Comment:
If it's not JSON, nothing is changed. If it is, it definitely won't work if we don't do any escaping.